apps/web/src/services/dashboards/widgets/asset-widgets/severity-status-by-service/SeverityStatusByServiceWidget.vue

```diff
@@ -38,48 +39,115 @@ import {
 } from 'vue';
 
 import dayjs from 'dayjs';
+import { min } from 'lodash';
+
+import { SpaceConnector } from '@cloudforet/core-lib/space-connector';
+import { ApiQueryHelper } from '@cloudforet/core-lib/space-connector/helper';
 
 import type { DateRange } from '@/services/dashboards/config';
 import WidgetFrame from '@/services/dashboards/widgets/_components/WidgetFrame.vue';
+import type { ComplianceStatus, Severity } from '@/services/dashboards/widgets/_configs/asset-config';
 import { SEVERITY_STATUS_MAP } from '@/services/dashboards/widgets/_configs/asset-config';
-import type { WidgetProps, WidgetExpose } from '@/services/dashboards/widgets/_configs/config';
+import type { WidgetExpose, WidgetProps } from '@/services/dashboards/widgets/_configs/config';
 import { useWidgetFrameProps } from '@/services/dashboards/widgets/_hooks/use-widget-frame-props';
 // eslint-disable-next-line import/no-cycle
 import { useWidgetLifecycle } from '@/services/dashboards/widgets/_hooks/use-widget-lifecycle';
 // eslint-disable-next-line import/no-cycle
 import { useWidgetState } from '@/services/dashboards/widgets/_hooks/use-widget-state';
 
 
+interface Data {
+    service: string;
+    complianceStatus: ComplianceStatus;
+    severity: Severity;
+    value: number;
+}
 const DATE_FORMAT = 'YYYY-MM';
 const SEVERITY_STATUS_MAP_VALUES = Object.values(SEVERITY_STATUS_MAP);
 const props = defineProps<WidgetProps>();
 const state = reactive({
-    ...toRefs(useWidgetState(props)),
+    ...toRefs(useWidgetState<Data[]>(props)),
     dateRange: computed<DateRange>(() => ({
         start: dayjs.utc(state.settings?.date_range?.start).format(DATE_FORMAT),
         end: dayjs.utc(state.settings?.date_range?.end).format(DATE_FORMAT),
     })),
-    sampleData: computed(() => [...Array(20)].map(() => Object.values(SEVERITY_STATUS_MAP)[Math.floor(Math.random() * 5)])),
     boxWidth: computed<number>(() => {
         if (!props.width) return 112;
         const widgetPadding = 24;
         const widgetContentWidth = props.width - (widgetPadding * 2);
         if (props.width >= 990) return widgetContentWidth / 8;
-        return widgetContentWidth / 7;
+        return widgetContentWidth / 7 < 112 ? 112 : widgetContentWidth / 7;
```

Comment:
question: What does number '112' mean? 😮

Reply:
It's min width of box! I'll replace it to constant value :)